File: schemas/project-management/drizzle/task_comments.ts
// task_comments: threaded discussion entries attached to tasks.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { tasks } from "./tasks";
import { users } from "./users";

export const taskComments = pgTable(
  "task_comments",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    taskId: uuid("task_id").notNull().references(() => tasks.id, { onDelete: "cascade" }),
    userId: uuid("user_id").references(() => users.id, { onDelete: "set null" }),
    parentId: uuid("parent_id").references(() => taskComments.id, { onDelete: "cascade" }),
    content: text("content").notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_task_comments_task_id").on(table.taskId),
    index("idx_task_comments_parent_id").on(table.parentId),
    index("idx_task_comments_user_id").on(table.userId),
  ]
);

File: schemas/project-management/spacetimedb/task_comments.rs
// task_comments: Threaded discussion comments on tasks.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = task_comments, public)]
pub struct TaskComment {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub task_id: String, // UUID — FK → tasks.id (cascade delete)
    #[index(btree)]
    pub user_id: Option<String>, // UUID — FK → users.id (set null)
    #[index(btree)]
    pub parent_id: Option<String>, // UUID — FK → task_comments.id (cascade delete)
    pub content: String,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
